src/main/java/redis/clients/jedis/BinaryJedisCluster.java

```diff
@@ -0,0 +1,1261 @@
+        return connection.zrevrangeByScoreWithScores(key, max, min, offset, count);
+      }
+    }.runBinary(key);
+  }
+
+  @Override
+  public Set<Tuple> zrevrangeByScoreWithScores(final byte[] key, final byte[] max,
+      final byte[] min, final int offset, final int count) {
+    return new JedisClusterCommand<Set<Tuple>>(connectionHandler, timeout, maxRedirections) {
+      @Override
+      public Set<Tuple> execute(Jedis connection) {
+        return connection.zrevrangeByScoreWithScores(key, max, min, offset, count);
+      }
+    }.runBinary(key);
+  }
+
+  @Override
+  public Long zremrangeByRank(final byte[] key, final long start, final long end) {
+    return new JedisClusterCommand<Long>(connectionHandler, timeout, maxRedirections) {
+      @Override
+      public Long execute(Jedis connection) {
+        return connection.zremrangeByRank(key, start, end);
+      }
+    }.runBinary(key);
+  }
+
+  @Override
+  public Long zremrangeByScore(final byte[] key, final double start, final double end) {
+    return new JedisClusterCommand<Long>(connectionHandler, timeout, maxRedirections) {
+      @Override
+      public Long execute(Jedis connection) {
+        return connection.zremrangeByScore(key, start, end);
+      }
+    }.runBinary(key);
+  }
+
+  @Override
+  public Long zremrangeByScore(final byte[] key, final byte[] start, final byte[] end) {
+    return new JedisClusterCommand<Long>(connectionHandler, timeout, maxRedirections) {
+      @Override
+      public Long execute(Jedis connection) {
+        return connection.zremrangeByScore(key, start, end);
+      }
+    }.runBinary(key);
+  }
+
+  @Override
+  public Long linsert(final byte[] key, final Client.LIST_POSITION where, final byte[] pivot,
+      final byte[] value) {
+    return new JedisClusterCommand<Long>(connectionHandler, timeout, maxRedirections) {
+      @Override
+      public Long execute(Jedis connection) {
+        return connection.linsert(key, where, pivot, value);
+      }
+    }.runBinary(key);
+  }
+
+  @Override
+  public Long lpushx(final byte[] key, final byte[]... arg) {
+    return new JedisClusterCommand<Long>(connectionHandler, timeout, maxRedirections) {
+      @Override
+      public Long execute(Jedis connection) {
+        return connection.lpushx(key, arg);
+      }
+    }.runBinary(key);
+  }
+
+  @Override
+  public Long rpushx(final byte[] key, final byte[]... arg) {
+    return new JedisClusterCommand<Long>(connectionHandler, timeout, maxRedirections) {
+      @Override
+      public Long execute(Jedis connection) {
+        return connection.rpushx(key, arg);
+      }
+    }.runBinary(key);
+  }
+
+  @Override
+  public List<byte[]> blpop(final byte[] arg) {
+    return new JedisClusterCommand<List<byte[]>>(connectionHandler, timeout, maxRedirections) {
+      @Override
+      public List<byte[]> execute(Jedis connection) {
+        return connection.blpop(arg);
+      }
+    }.runBinary(arg);
+  }
+
```

Comment:
@christophstrobl 
Thanks for reporting!
First of all, blpop(byte[] arg) should be removed, and in reviewing now via #902.
Regarding to blpop(byte[] arg, int timeout), we should add List<byte[]> blpop(byte[] arg, int timeout) to BinaryJedisCommands in order to add method.